### PR TITLE
SARS-Cov-2 Support

### PR DESCRIPTION
**Pull request by jvwong**
Introduces three key pieces:

1. Wrapper for NCBI EUTILS
  - Basically copy-paste from Factoid

2. Index proteins for a given taxon in `datasource.js/ncbi`
  - Add SARS-CoV-2 genes from file using existing code
  - Introduce protein-level records for SARS-CoV-2 due due to multiple mature protein chains derived from genes
  - The complement of proteins is sourced from NCBI EUTILS, given an organisms ID

3. Indexing `processing.js`
  - Index the SARS-CoV-2 records all at once from 'source' i.e, not a file

Notes:
  - Overall grounding tests did not deviate much at all (~9% error rate) from before (analyzed about a year ago?)
  - For proteins, synonyms are generated based upon the title/label which have some particular formatting in NCBI
  - There's probably no consensus on synonyms or naming given how new this all is
  - Need to add env variable for `NCBI_EUTILS_API_KEY`
  

Refs #72, https://github.com/PathwayCommons/factoid/issues/699

### CODE REVIEW

**Comment by jvwong** [comment]
Sample usage in Factoid:

- [Structure of the SARS-CoV-2 spike receptor-binding domain bound to the ACE2 receptor Nature 2020](https://www.nature.com/articles/s41586-020-2180-5)



- [A SARS-CoV-2 protein interaction map reveals targets for drug repurposing Nature 2020](https://www.nature.com/articles/s41586-020-2286-9)

File: src/server/config.js [hunk 1]
```diff
@@ -38,3 +38,6 @@ export const CHEBI_URL = env('CHEBI_URL', 'ftp://ftp.ebi.ac.uk/pub/databases/che
 
 export const NCBI_FILE_NAME = env('NCBI_FILE_NAME', 'ncbi');
 export const NCBI_URL = env('NCBI_URL', 'ftp://ftp.ncbi.nih.gov/gene/DATA/gene_info.gz');
+export const NCBI_EUTILS_BASE_URL = env('NCBI_EUTILS_BASE_URL', 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/');
+export const NCBI_EUTILS_API_KEY = env('NCBI_EUTILS_API_KEY', 'b99e10ebe0f90d815a7a99f18403aab08008');
```
Comment (maxkfranz) [inline src/server/config.js:42]: Is this a non-prod key?

**Comment by jvwong** [comment]
> Looks good. Some points:
> 
>     * It's probably not worth it to handle the genes right now, so just cherry-picking the proteins makes sense from a risk-reward perspective.
>

The genes are in there, simply by virtue of adding the Organism. 
 
>     * How risky is it to use Etutils HTTP requests for indexing versus FTP?
> 

I can look around. Maybe there's a particular download file that's small. The EUTIL JSON is tiny (38 JSON objects)

>     * At some point, there should be tests for sars-cov-2.  Right now, experimentation on a live instance is more important, so we should just merge as-is.

Yeah, I think there may be a lot of noise given how new it all is.

**Comment by maxkfranz** [comment]
> I can look around. Maybe there's a particular download file that's small. The EUTIL JSON is tiny (38 JSON objects)

If that's the case, then it's probably fine as-is.  I'm mostly concerned about failed requests, since Eutils can be finicky.  As long as indexing errors are salient, it's manageable for now:  We can just rebuild again, at least for prod instances.

**Comment by maxkfranz** [comment]
TODO: Update the staging VM with the Etutils key env var.

**Comment by jvwong** [comment]
> TODO: Update the staging VM with the Etutils key env var.

- Update README.md here
- Update docs @ github.com/BaderLab/sysadmin/
- Check unstable factoid is picking up the new org

**Comment by maxkfranz** [comment]
We may reconsider patch files (#41) to accommodate poorly labelled entries (e.g. spike protein), if there isn't a better datasource.

**Pull request merged**